core/src/main/scala/org/apache/spark/resource/TaskResourceRequests.scala

```diff
@@ -37,6 +38,8 @@ private[spark] class TaskResourceRequests() extends Serializable {
 
   def requests: Map[String, TaskResourceRequest] = _taskResources.asScala.toMap
 
+  def requestsJMap: JMap[String, TaskResourceRequest] = _taskResources.asScala.asJava
```

Comment:
@tgravescs . To return immutable map, `asScala.toMap.asJava` is required.

Reply:
yes